src/Common/src/System/Net/CookieParser.cs

```diff
@@ -667,6 +667,13 @@ internal Cookie Get()
                                         {
                                             cookie.Expires = expires;
                                         }
+                                        else if (HttpDateParser.TryStringToDate(CheckQuoted(_tokenizer.Value), out DateTimeOffset expiresUtc))
```

Comment:
`HttpDateParser.TryStringToDate` will try a wide variety of common formats, and output RFC 1123 style dates. However, the [original issue](https://github.com/dotnet/corefx/issues/27861#issuecomment-392171121) is about only accept cookies with UTC timezone. Since you [only verified the UTC case](https://github.com/dotnet/corefx/issues/27861#issuecomment-392171121) (other formats are not tested), I'm concerned that this change is too much. Maybe we should not use `HttpDateParser` here.

https://github.com/dotnet/corefx/blob/31fa4fcc5fcb12a43a8303883fbf9affe0746293/src/Common/src/System/Net/HttpDateParser.cs#L11-L35

cc: @davidsh

Reply:
In thinking about this more, perhaps the safest change to make is not here in the HTTP stack.

Instead, the DateTime.TryParse() method should be changed to allow for "UTC" timezone in addition to "GMT".  Both "UTC" and "GMT" should be semantically equivalent.

I think the DateTime class is in the CoreClr repo.